src/lib/transformers/tokend.js

```diff
@@ -171,12 +149,18 @@ class TokendTransformer {
           return Promise.resolve(Immutable.Map().setIn(keyPath, null));
         }
 
-        this._cache[signature] = data;
+        this._cache[propertyName] = {
+          signature,
+          plaintext: data.plaintext
+        };
 
         return Promise.resolve(Immutable.Map().setIn(keyPath, data.plaintext));
       }).catch((err) => {
         Log.log('WARN', err);
         this._client.clearCacheAtKey(method, requestId);
+        if (this._cache.hasOwnProperty(propertyName)) {
+          return Promise.resolve(Immutable.Map().setIn(keyPath, this._cache[propertyName].plaintext));
+        }
 
         return Promise.resolve(Immutable.Map().setIn(keyPath, null));
```

Comment:
Might be worth adding some logging to surface when there's an error from Tokend and the property doesn't exist in the cache.

Reply:
All set, added a Log.log(`Warn`) message after the if block